### ISSUE TITLE
New check for "Explicit type argument can be replaced with <>"

### ISSUE THREAD

**Issue by mmoayyed** [checkstyle/checkstyle#2943]
So this:

``` java
Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<String, Object>());
```

Becomes:

``` java
Map<String, Object> EMPTY_MAP = Collections.unmodifiableMap(new HashMap<>());
```
##

**Comment by romani** [comment]
Already implemented at sevntu project, but still have false positives. https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues?q=is%3Aissue+DiamondOperatorForVariableDefinitionCheck+is%3Aopen

You are welcome with PullRequests.

**Comment by romani** [comment]
lets close the issue, as moratorium finish we could start moving all from sevntu.

**Comment by AustinShalit** [comment]
Just checking in to see if anyone was working on this issue.

**Comment by rnveach** [comment]
@AustinShalit Issue is closed and check is already implemented in sevntu.
There is no plans to move checks at this time, especially since we are in the middle of GSoC.

**Comment by romani** [comment]
@AustinShalit , if you have a time, you can move it to main project. We will guide you on the process and testing that is required. Please let me know if you are interested.

**Comment by AustinShalit** [comment]
I will gladly add it in.  Just let me know what the requirements are.

**Comment by rnveach** [comment]
@AustinShalit Once you add the check to checkstyle's code. Our unit tests and CI will guide you on pretty much everything that is needed.
Checkstyle will require documentation for it in our xdocs. We may have more strict coding rules, so changes may be required to the check. We also have mutation testing, so you may be required to add more tests.
We will guide you once a PR is started and we see what is actually happening.

**Comment by AustinShalit** [comment]
Can a maintainer please approve this issue for pull request #4858?

**Comment by rnveach** [comment]
Done.

**Comment by nitsshukla** [comment]
How to ignore this check?
I'm getting the warning but upon removing it gets compile-time error.
`private static final Map<String,String> NAMES = new HashMap<String, String>(){{
        put("#timestamp", "timestamp");
        put("#status", "status");
        put("#path", "path");
    }};`

**Comment by AustinShalit** [comment]
@nitsshukla You are in the wrong spot.  This feature was never merged into checkstyle due to some corner cases.

**Comment by nitsshukla** [comment]
> @nitsshukla You are in the wrong spot. This feature was never merged into checkstyle due to some corner cases.

Oh ok, thanks.

**Comment by vlsi** [comment]
I'm inclined the check should be auto-correctable, so OpenRewrite or IDEA are better tools than Checkstyle here.

See https://docs.openrewrite.org/reference/recipes/java/cleanup/usediamondoperator
